### DISCUSSION TITLE
[Bug] The reactive range of hovering of Markers is larger than the underlining range

### DISCUSSION THREAD

**Discussion by chengtie** [microsoft/monaco-editor#3704] [Q&A]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Code

```typescript
monaco.languages.register({ id: 'myLanguage' });

monaco.editor.onDidCreateModel(function(model) {
      var markers = [{
        severity: monaco.MarkerSeverity.Error,
        startLineNumber: 1,
        startColumn: 3,
        endLineNumber: 1,
        endColumn: 4,
        message: 'need to correct c'
      },
      {
        severity: monaco.MarkerSeverity.Error,
        startLineNumber: 1,
        startColumn: 4,
        endLineNumber: 1,
        endColumn: 5,
        message: 'need to correct d'
      }];
      monaco.editor.setModelMarkers(model, 'myLanguage', markers);
});

var ed = monaco.editor.create(document.getElementById("container"), {
    value: "abcdefg",
    language: "myLanguage",
    lightbulb: { enabled: false },
})
```

### Actual Behavior

1- If I hover on `c` (or put the focus between `b` and `c`, then press `Ctrl+K Ctrl+I`), both of the two messages are shown. 

2- Similarly, if I hover on `d` (or put the focus between `d` and `e`, then press `Ctrl+K Ctrl+I`), both of the two messages are shown.

### Expected Behavior

At Location 1, only the message for `c` is relevant and expected.

At Location 2, only the message for `d` is relevant and expected.

### Additional Context

In Monaco Editor, [Markers](https://microsoft.github.io/monaco-editor/api/interfaces/monaco.editor.IMarkerData.html) allow to underline a text and display a message when hovering over it. But I just realized that the reactive range of hovering is a little bit larger than the underlining range. As a consequence, messages can be confusing when two markers are adjacent.

**Comment by hediet** [comment]
I think this is by design - when a cursor touches two columns, both their hover message is shown.

**Comment by chengtie** [comment]
> I think this is by design - when a cursor touches two columns, both their hover message is shown.

Thank you. Is there a way to customize or modify this setting?

**Comment by hediet** [comment]
Not that I know of.
